worker/src/RTC/IceServer.cpp

```diff
@@ -370,22 +384,29 @@ namespace RTC
 					SetSelectedTuple(storedTuple);
 					// Update state.
 					this->state = IceState::CONNECTED;
+					// Update nomination.
+					this->remoteNomination = nomination;
 					// Notify the listener.
 					this->listener->OnIceServerConnected(this);
 				}
 				else
 				{
-					MS_DEBUG_TAG(ice, "transition from state 'disconnected' to 'completed'");
-
 					// Store the tuple.
 					auto* storedTuple = AddTuple(tuple);
 
-					// Mark it as selected tuple.
-					SetSelectedTuple(storedTuple);
-					// Update state.
-					this->state = IceState::COMPLETED;
-					// Notify the listener.
-					this->listener->OnIceServerCompleted(this);
+					if (nomination > this->remoteNomination)
+					{
+						MS_DEBUG_TAG(ice, "transition from state 'disconnected' to 'completed'");
+
+						// Mark it as selected tuple.
+						SetSelectedTuple(storedTuple);
```

Comment:
Wrong tabulation here right?

Reply:
Sorry. what do you mean?

Reply:
Sorry, wrong display in GH mobile. It's ok.